activerecord/lib/active_record/autosave_association.rb

```diff
@@ -176,7 +176,8 @@ def define_non_cyclic_method(name, &block)
         # check if the save or validation methods have already been defined
         # before actually defining them.
         def add_autosave_association_callbacks(reflection)
-          save_method = :"autosave_associated_records_for_#{reflection.name}"
+          method_suffix = reflection.options[:class_name].try(:underscore)
```

Comment:
I've also created a benchmark and saw that using join is faster than my approach. I'll go with it thanks.